### PR TITLE
[DEN-316] origin v3 group create, read and delete implementation

### PR DESCRIPTION
**Pull request by ssaluja4**

### CODE REVIEW

File: edgecast/resources/originv3/origin_schema.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
+// Copyright 2022 Edgecast Inc., Licensed under the terms of the Apache 2.0 license.
+// See LICENSE file in project root for terms.
+package originv3
+
+import "github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+func GetOriginV3Schema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
```
Comment (stevenpaz) [inline edgecast/resources/originv3/origin_schema.go:8]: We need a `platform` schema argument to determine which endpoint to call.

**Review by stevenpaz — CHANGES_REQUESTED** [review]

File: edgecast/helper/collections.go [hunk 2]
```diff
@@ -37,6 +37,26 @@ func ConvertSliceToStrings(v []interface{}) ([]string, error) {
 	return strings, nil
 }
 
+// ConvertSliceToStrings converts a []interface{} to []string.
```
[the remark this answers is not in the document]
Comment (ssaluja4) [reply edgecast/helper/collections.go]: updated.

File: edgecast/helper/tf.go [hunk 3]
```diff
@@ -124,6 +124,23 @@ func ConvertTFCollectionToStrings(v interface{}) ([]string, error) {
 	return nil, err
 }
 
+// ConvertTFCollectionToPtrStrings converts Terraform's
+// TypeList and TypeSet collections into a []string.
```
[the remark this answers is not in the document]
Comment (ssaluja4) [reply edgecast/helper/tf.go]: updated.

File: edgecast/provider.go [hunk 4]
```diff
@@ -143,6 +144,7 @@ func buildResourcesMap() map[string]*schema.Resource {
 		"edgecast_waf_scopes":             waf.ResourceScopes(),
 		"edgecast_waf_bot_rule_set":       waf.ResourceBotRuleSet(),
 		"edgecast_cps_certificate":        cps.ResourceCertificate(),
+		"edgecast_originv3":               originv3.ResourceOriginV3(),
```
[the remark this answers is not in the document]
Comment (ssaluja4) [reply edgecast/provider.go]: updated to reflect a separate httplarge resource

File: edgecast/resources/originv3/origin.go [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+		originv3.CustomerOriginGroupHTTPRequest{
+			Name:               originGroupState.Name,
+			HostHeader:         originGroupState.HostHeader,
+			ShieldPops:         originGroupState.ShieldPops,
+			NetworkTypeId:      originGroupState.NetworkTypeId,
+			StrictPciCertified: originGroupState.StrictPciCertified,
+			TlsSettings:        originGroupState.TlsSettings,
+		}
+
+	cresp, err := svc.HttpLargeOnly.AddHttpLargeGroup(cparams)
```
[the remark this answers is not in the document]
Comment (ssaluja4) [reply edgecast/resources/originv3/origin.go]: N/A separate resources for adn/httplarge

File: edgecast/resources/originv3/origin_schema.go (hunk 1 above)
Comment (ssaluja4) [reply edgecast/resources/originv3/origin_schema.go]: N/A separate resources for adn/httplarge

**Review by stevenpaz — APPROVED** [review]

**Pull request merged**